packages/start/src/server/defaultStreamHandler.tsx

```diff
@@ -16,30 +16,54 @@ export const defaultStreamHandler: StartHandler<AnyRouter> = async ({
   router,
   responseHeaders,
 }) => {
-  const passthrough = new PassThrough()
-
-  const pipeable = renderToPipeableStream(<StartServer router={router} />, {
-    ...(isbot(request.headers.get('User-Agent'))
-      ? {
-          onAllReady() {
-            pipeable.pipe(passthrough)
-          },
-        }
-      : {
-          onShellReady() {
-            pipeable.pipe(passthrough)
-          },
-        }),
-    onShellError(err) {
-      throw err
-    },
-  })
-
-  return new Response(passthrough as any, {
-    status: router.state.statusCode,
-    statusText: `${router.state.statusCode}`.startsWith('5')
-      ? 'Internal Server Error'
-      : 'OK',
-    headers: responseHeaders,
-  })
+  if (typeof ReactDOMServer.renderToReadableStream === 'function') {
```

Comment:
`renderToReadableStream` now has precedence over `renderToPipeableStream`, do we want this?

Reply:
For environments where `renderToReadableStream` is available (environment supporting Web Streams API i.e. the browser, Bun, etc.), `renderToPipeableStream` is not available.

See:

https://github.com/facebook/react/blob/main/packages/react-dom/package.json#L60-L70
https://github.com/facebook/react/blob/main/packages/react-dom/server.browser.js
https://github.com/facebook/react/blob/main/packages/react-dom/server.node.js
https://github.com/facebook/react/blob/main/packages/react-dom/server.edge.js
https://github.com/facebook/react/blob/main/packages/react-dom/server.bun.js

Reply:
ah, it's either - or. I assumed that an implementation could have both.